### ISSUE TITLE
11. Noyau

### ISSUE THREAD

**Issue by Vayel** [ArnaudCalmettes/cours-python-systeme#1]
> Simplement, vous le savez comme moi, un ordinateur, c'est d'abord tout un tas de composants matériels soudés sur des cartes et connectés entre eux par des bus de données.

Le lecteur est-il censé connaître la notion de bus ?

> Pour qu'un programme utilise un périphérique matériel, **il doit forcément faire appel à du code qui sert à manipuler ce matériel.**

Je ne comprends pas le "forcément", vu qu'on ne fait justement pas ça, je crois, au début de la partie "Interagir avec le matériel. Oui, mais…". 

Ou alors, peut-être ai-je mal compris cette phrase, et l'ai interprétée comme "*il doit forcément faire appel à du code externe" ? Le cas échéant, peut-être pourrait-on remplacer "faire appel à" par "exécuter" ?

> ils n'ont même pas besoin de mettre le nez dans la documentation technique, ils n'ont plus qu'à se concentrer sur leurs programmes à eux.

Il faut tout de même être capable d'utiliser le pilote. Du coup, peut-être faudrait-il préciser qu'on parle ici de la doc du disque ?

> En fait, de nos jours, les programmes n'interagissent jamais directement avec le matériel, sauf à la rigueur sur certains systèmes embarqués.

Là, je me suis dit "oui, parce qu'ils passent par des pilotes", mais j'imagine que ce n'est pas ce que tu sous-entends (puisque tu introduis le noyau) ?

> Vous l'avez compris : un système d'exploitation digne de ce nom doit fournir aux programmes une interface commune, de façon à ce que ceux-ci n'aient jamais besoin de se soucier du matériel sur lequel ils agissent.

Je le mets là, parce que je me suis fait la remarque à ce moment : ne penses-tu pas qu'il faudrait introduire le lecteur à la notion d'OS ? En effet, la personne ignorant que l'objectif de ce dernier est d'offrir une abstraction au matériel ne comprendra probablement pas trop cette phrase, laquelle reviendrait à "Windows doit fournir aux programmes une interface commune", en ignorant ce qu'est réellement Windows (outre le truc qui se lance quand j'allume ma machine).

Mais peut-être est-ce hors-propos ?

> HAL va faire bien attention à trouver une adresse libre sur le matériel pour écrire ces données.

Histoire de bien insister sur le rôle de l'HAL, peut-être pourrait-on détailler un peu plus le processus. Je ne suis pas connaisseur en la matière, mais j'imagine qu'il y a au moins l'étape "chercher le pilote approprié".

En fait, tu le fais avec la figure 1,3. ^^

> Les appels systèmes en eux-mêmes sont réalisés au moyen d'une interruption matérielle.

Le lecteur est-il censé connaître la notion d'interruption matérielle, ou est-ce juste un point de vocabulaire. Dans le premier cas, ça me semble un pré-requis assez important, si on le compare au contenu de cette partie. Dans le second, c'est frustrant de ne pas avoir plus d'informations, d'ignorer ce qu'*est* une interruption matérielle.

> Cas des Unixoïdes : TODO

Très bon boulot en tout cas.

**Comment by ArnaudCalmettes** [comment]
> Le lecteur est-il censé connaître la notion de bus ?

Non. Celle d'interruption matérielle non plus, mais ça n'a aucune inportance. Je rajouterai des liens Wikipedia pour ces notions périphériques.

> Je ne comprends pas le "forcément", vu qu'on ne fait justement pas ça, je crois, au début de la partie "Interagir avec le matériel. Oui, mais…".

Faire appel à du code, ça signifie simplement "appeler du code". En toute rigueur rien ne dit que ce code est externe ou même codé dans une fonction différente du programme.

> ne penses-tu pas qu'il faudrait introduire le lecteur à la notion d'OS ?

Absolument, et c'est plus ou moins ce que fait tout le premier chapitre... Je suppose que tu veux dire "définir en une phrase", pourquoi pas, mais ça n'apporterait pas grand chose au propos parce que cette phrase serait soit trop longue, soit trop vague.

> Comment trouve-t-on ce numéro ?

1. En allant lire le code-source de la version de linux installée sur la machine.
2. En traçant les appels systèmes réalisés par un programme qui y a recours.
3. En desassemblant un programme en C qui réalise cet appel.

... En somme, en suivant une démarche beaucoup trop compliquée pour en parler dès le premier chapitre.

>  Le "Ces" désigne-t-il tous les groupes, ou seulement les secondaires ?

Tous.

> J'imagine que le GID n'est pas nécessairement égal au UID ?

Non, évidemment, sinon il n'y aurait pas besoin de définir 2 valeurs.

>  Peut-être préciser qu'on est revenu à Python, même si le code en-dessous le montre ?

?? 

On ne quitte jamais Python.

**Comment by Vayel** [comment]
> ??
> 
> On ne quitte jamais Python.

On fait tout de même un peu de shell.
